baseplate/lib/message_queue.py

```diff
@@ -111,14 +147,100 @@ def unlink(self) -> None:
+    that don't have Posix available. The client will
+    instantitate a RemoteMessageQueue, which connect to
+    a Thrift server. The Thrift server internally uses an
+    InMemoryMessageQueue to store data.
+
+    """
+
+    def __init__(self, name: str, max_messages: int):
+        self.queue: q.Queue = q.Queue(max_messages)
+        self.max_messages = max_messages
+        self.name = name
+
+    def get(self, timeout: Optional[float] = None) -> bytes:
+        try:
+            message = self.queue.get(timeout=timeout)
+            self.queue.task_done()
+            return message
+        except q.Empty:
+            raise TimedOutError
+
+    def put(self, message: bytes, timeout: Optional[float] = None) -> None:
+        try:
+            self.queue.put(message, timeout=timeout)
+        except q.Full:
+            raise TimedOutError
+
+    def close(self) -> None:
+        """Not implemented for in-memory queue"""
+
+
+class RemoteMessageQueue(MessageQueue):
+    """A message queue that uses a remote Thrift server.
+
+    Used in conjunction with the InMemoryMessageQueue to
+    provide an alternative to Posix queues, for systems
+    that don't have Posix available.
+
+    """
+
+    def __init__(self, name: str, max_messages: int, host: str = "127.0.0.1", port: int = 9090):
+        # Connect to the remote queue server, and creeate the new queue
+        self.name = name
+        self.max_messages = max_messages
+        self.host = host
+        self.port = port
+        self.connect()
+        self.client.create_queue(name, max_messages)
+
+    def connect(self) -> None:
+        # Establish a connection with the queue server
+        transport = TSocket.TSocket(self.host, self.port)
+        self.transport = TTransport.TBufferedTransport(transport)
+        protocol = TBinaryProtocol.TBinaryProtocol(self.transport)
+        self.client = RemoteMessageQueueService.Client(protocol)
+        self.transport.open()
+
+    def get(self, timeout: Optional[float] = None) -> bytes:
+        # Call the remote server and get an element for the correct queue
+        try:
+            try:
+                return self.client.get(self.name, timeout).value
+            except TSocket.TTransportException:
+                # Try reconnecting once, we dont want this as another top-level except because
+                # we may get a timeout after re-connecting, and we want to catch that
+                self.connect()
+                return self.client.get(self.name, timeout).value
+        # If the server responded with a timeout, raise our own timeout to be consistent with the posix queue
+        except ThriftTimedOutError:
+            raise TimedOutError
+
+    def put(self, message: bytes, timeout: Optional[float] = None) -> None:
+        # Call the remote server and put an element on the correct queue
+        # Will create the queue if it doesnt exist
+        try:
+            try:
```

Comment:
Same as the get above I would use a private wrapper function to avoid a nested try. It is a style thing but nested trys have a habbit of getting deeper and messier over time so I alway try to avoid them for readability sake. Up to you though.